presto-main/src/main/java/io/prestosql/execution/SqlQueryManager.java

```diff
@@ -313,4 +326,24 @@ private void enforceCpuLimits()
             }
         }
     }
+
+    /**
+     * Enforce max raw data size at the query level
+     */
+    public void enforceQueryMaxInputDataSizeLimits()
+    {
+        for (QueryExecution query : queryTracker.getAllQueries()) {
+            Optional<DataSize> limitOpt = Optional.ofNullable(getQueryMaxDataSize(query.getSession())
+                    .flatMap(sessionLimit -> maxQueryInputDataSize.map(serverLimit -> Ordering.natural().min(serverLimit, sessionLimit)))
+                    .orElse(maxQueryInputDataSize.orElse(null)));
+
+            limitOpt.ifPresent(limit -> {
+                DataSize processedInputDataSize = query.getQueryInfo().getQueryStats().getProcessedInputDataSize();
```

Comment:
What's the rationale of looking at `processedInputDataSize` here instead of `physicalInputDataSize` ?
If we block queries based on the physical data size read from data source, it's easier for users to look at the sizes of their tables and come up with a reasonable limit based on that. Uncompressed data size seems harder to reason about when defining limits.
There is another PR https://github.com/prestosql/presto/pull/4076 which is trying to limit by `rawInputDataSize`.

Reply:
There is no rational: `query.getBasicQueryInfo().getQueryStats().getRawInputDataSize();` was used at first.
I still haven't been able to find a clean way to write a test, the only way I found was to extract this specific piece of code `query.getBasicQueryInfo().getQueryStats().getRawInputDataSize();` into a dedicated function to be able to mock it in the test.
The current state was to illustrate the issues I have writing a test for it. I'll repush a earlier version without the test + addressing Praveen's comment.